packages/ember-htmlbars/lib/node-managers/view-node-manager.js

```diff
@@ -27,13 +27,9 @@ function ViewNodeManager(component, scope, renderNode, block, expectElement) {
 export default ViewNodeManager;
 
 ViewNodeManager.create = function ViewNodeManager_create(renderNode, env, attrs, found, parentView, path, contentScope, contentTemplate) {
-  assert('HTMLBars error: Could not find component named "' + path + '" (no component or template with that name was found)', function() {
```

Comment:
Instead of rewriting to avoid the function, can you make the function an IIFE instead?  I personally find the function style of writing this much easier to understand.